Pytorch documentation fix and update

luigibonati opened the pull request:
##### Description

Small bug fixes and updates related to the Pytorch module documentation.

- I noticed that there was a bug in the documentation of the pytorch module. Due to recent changes in section headers, only the first two sections were shown.
- I also updated some instructions on how to verify that the configuration with libtorch is succesful. 
- I updated the module doc and the CI instructions with the link to the latest libtorch version (1.13.1). Before the Long Term Support version (1.8.2) was recommended and used in the CI, but it has been [discontinued](https://pytorch.org/blog/pytorch-enterprise-support-update/).
- I updated the behavior of the PLUMED_CHECK_CXX_PACKAGE function in configure  when linking against multiple libraries. Now the check is done against all the libraries at once, rather than trying by adding one at a time (which was also creating confusion in the config.log)

##### Target release

I would like my code to appear in release 2.9.

##### Type of contribution

- [X] changes to code or doc authored by PLUMED developers, or additions of code in the core or within the default modules
- [ ] changes to a module not authored by you
- [X] new module contribution or edit of a module authored by you

##### Copyright

- [X] I agree to transfer the copyright of the code I have written to the PLUMED developers or to the author of the code I am modifying.

- [X] the module I added or modified contains a `COPYRIGHT` file with the correct license information. Code should be released under an open source license. I also used the command `cd src && ./header.sh mymodulename` in order to make sure the headers of the module are correct. 

##### Tests

- [ ] I added a new regtest or modified an existing regtest to validate my changes.
- [] I verified that all regtests are passed successfully on [GitHub Actions](https://github.com/plumed/plumed2/actions).

GiovanniBussi commented:
@luigibonati thanks! A couple of points before merging:
- Can you confirm that the change to configure.ac only affects the tests done for libtorch? I think that all other calls to `PLUMED_CHECK_CXX_PACKAGE` are missing the fifth argument (`multiple_libraries`), but I wanted to double check that I understood correctly.
- I noticed you are also using a different autoconf version (I guess the ubuntu ones). I can fix this after merge. There was a problem with another autoconf version recently (#897) so I try to stick to the current one.

luigibonati commented:
> @luigibonati thanks! A couple of points before merging:
> 
> * Can you confirm that the change to configure.ac only affects the tests done for libtorch? I think that all other calls to `PLUMED_CHECK_CXX_PACKAGE` are missing the fifth argument (`multiple_libraries`), but I wanted to double check that I understood correctly.

Yes, exactly. I had added the optional argument multiple_libraries to allow linking to more than one library at the same time, instead of trying each one individually as was the case with the other libraries. So this change will only involve libtorch. In any case, the behavior is the same, I just simplified it by having the test run with all the specified libraries, instead of first trying only the first library, then the first and second, and so on.

> * I noticed you are also using a different autoconf version (I guess the ubuntu ones). I can fix this after merge. There was a problem with another autoconf version recently ([Centos 7 and rt-make-c11 #897](https://github.com/plumed/plumed2/issues/897)) so I try to stick to the current one.

Sounds good! yes I used the ubuntu one
